### PR TITLE
Tools localization

### PR DESCRIPTION
**Pull request by makiam**
Tools module localization. Only default, finnish, russian and empty english properties files added

### CODE REVIEW

**Comment by Sailsman63** [comment]
Why name the resource file "Bundle"? Seems a bit to generic and non-indicative.

**Comment by makiam** [comment]
I see such naming convention inside Netbeans sources. Every sub-project stores it strings into Bundle.properties files.
IMO the name for this properties file is not important as resource accessed via registered resource id

**Review by Sailsman63 — CHANGES_REQUESTED** [review]
Overall, I'm also wondering why all of the periods in the keys? They don't really seem to be used for namespacing...

**Comment by Sailsman63** [comment]
But AOI has a widely spread convention of naming the resource file something meaningful, even if it is not technically required. I agree that the resource ID is more important, and in this case would really prefer that it be `Tools` rather than `stdtools`. Do you have a reason to make it otherwise?

File: Tools/src/Bundle.properties [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+tube.dialog.name=Tube
+tube.width.label=Tube Width
+tube.cap.ends.label=Cap Ends
+tube.cap.end.open=Open Ends
+tube.cap.end.flat=Flat Ends
+skin.dialog.title=Skin
+button.skin.dialog.move.up=Move Up
+button.skin.dialog.move.down=Move Down
+skin.dialog.reverse=Reverse Direction
+skin.tool.message.curves=ou must select two or more curves to create a skin across.
```
[the remark this answers is not in the document]
Comment (makiam) [reply Tools/src/Bundle.properties]: I add dots just for readability

File: Tools/src/Bundle_fi.properties [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+extrude.target.label=Object to Extrude:
+extrude.direction.label=Extrude Direction:
+extrude.xaxis=X
+extrude.yaxis=Y
+extrude.zaxis=Z
+extrude.segments.label=Segments:
+extrude.twist.value.label=Twist (degrees):
+extrude.surface.accuracy.label=Surface Accuracy:
+extrude.follws.curve.label=Orientation Follows Curve
+csg.tool.message=You must select two objects for boolean modelling, at least one of which must be solid.
```
Comment (Sailsman63) [inline Tools/src/Bundle_fi.properties:57]: Why is there a Finnish properties file with all English localizations?

Comment (Sailsman63) [reply Tools/src/Bundle_fi.properties]: Okay, I see partial translations. Untranslated strings should not be present at all, though.

Comment (makiam) [reply Tools/src/Bundle_fi.properties]: Removed

File: Tools/src/Bundle.properties (hunk 1 above)
[the remark this answers is not in the document]
Comment (Sailsman63) [reply Tools/src/Bundle.properties]: Did you test this?

I'm getting the (untranslated) keys when I view this in English. For better or worse, the current code in `artofillusion.ui.Translate` expects dots to delimit semantic prefixes or suffixes. See the image:  

ArrayTool

**Review by peteihis — CHANGES_REQUESTED** [review]
Could you just leave the untranslated lines there of if you must do something just mark them as comment lines?

It is much easier to do the translation, when all that needs translating is already present in one file. Otherwise who ever is working on it, needs to keep comparing two files and looking for any differences. Most of the energy goes into hunting down the differences rather than actually working on the translations itself.

I'd prefer this approach to the rest of the AoI translations as well. Currently there are a lot of new untranslated texts scattered all over the translation files. It would be relatively easy to build a script that marks all the new entries (or removed ones) in all of the files. I think, in its tiny part that just _might_ help to keep the translation files better up to date.

File: Tools/src/Bundle.properties (hunk 1 above)
[the remark this answers is not in the document]
Comment (makiam) [reply Tools/src/Bundle.properties]: I forget to update Tools build script to package properties resources to jar file.
Can You please try again with last commit?

**Review by Sailsman63 — APPROVED** [review]

**Review by peteihis — COMMENTED** [review]
Thank you : )

**Comment by Sailsman63** [comment]
> mark them as comment lines?

> It is much easier to do the translation, when all that needs translating is already present in one file.

The flip side of this is that you now have a disconnected version of the english translation that may or may not be in sync with the current standard. It's a false sense of security, and you probably should be checking against the primary file anyway.

**Comment by Sailsman63** [comment]
Okay, we'll give it a try this way.

**Pull request merged**